### PR TITLE
Add definitions and mappings for REMIND 3.2

### PR DESCRIPTION
**Pull request by danielhuppmann**

### CODE REVIEW

**Comment by phackstock** [comment]
Ah that's different to iamconsortium/nomenclature#319 though. This one's failing because I cleaned up the CLI and adjusted it to the standard of giving multiple input values.
From:
```console
nomenclature validate-project . --dimensions "['region', 'scenario', 'variable']"
```
you'll now have to do:
```console
nomenclature validate-project . --dimensions region --dimensions scenario --dimensions variable
```

Reason for the change was that the first solution was a custom one from us, the current one adheres to the standards of CLI programs. I could think about offering a shorthand `-d` to make the command shorter.

**Comment by robertpietzcker** [comment]
I am sorry, I didn't think before when replying to the email...
is this supposed to be a different REMIND version than the one Renato used before? I think not, or do you disagree, @Renato-Rodrigues  ?

**Comment by danielhuppmann** [comment]
> is this supposed to be a different REMIND version than the one Renato used before?

No scenarios were submitted to the ECEMF-internal Scenario Explorer from REMIND 3.2 yet. This PR is just a fix such that native-region-names are "REMIND 3.2|..." instead of "REMIND 3.0|...". The mapping assumes the 21-region-version of REMIND, simplifying the mappings.

Please review that my allocation of the non-EU European countries to the iso3_codes-attribute are correct - I copied them from the REMIND 3.1 definitions at https://github.com/IAMconsortium/common-definitions/blob/main/definitions/region/native_regions/REMIND_3.1.yml.

I also noticed that REMIND submits already-aggregated for "EU27" in some project - are you also planning to do this for REMIND 3.2 in the ecemf-internal instance?

**Comment by Renato-Rodrigues** [comment]
> No scenarios were submitted to the ECEMF-internal Scenario Explorer from REMIND 3.2 yet.

There was no change in the regions definitions from REMIND 3.0 to REMIND 3.1 or REMIND 3.2. 
Therefore, a few 3.2 results were already uploaded using the previous mapping.
So my question is why did you not copied directly the below mapping? 
https://github.com/IAMconsortium/common-definitions/blob/main/mappings/REMIND_3.1.yml

> The mapping assumes the 21-region-version of REMIND, simplifying the mappings.

REMIND results can be uploaded in both 12-regions and 21-regions configurations, so I would not remove the support to the 12 regions (EUR and NEU).

> Please review that my allocation of the non-EU European countries to the iso3_codes-attribute are correct

They should be correct, but I will double check them soon.

> I also noticed that REMIND submits already-aggregated for "EU27" in some project - are you also planning to do this for REMIND 3.2 in the ecemf-internal instance?

Yes, we handle ourselves our EU27 regions aggregations and upload them directly.

One last point, any special reason why you removed the support for the R5 and R10 automatic aggregation of AR6 regions from the 3.1 common-definitions mapping?

**Comment by danielhuppmann** [comment]
Note that ECEMF does not use R9 and R10 region definitions (yet?), that's why I started from https://github.com/openENTRANCE/openentrance/blob/main/mappings/remind_3.0.yaml, not the mapping in common-definitions. In turn, the common-definitions repository does not have the INNOPATHS mappings...

Also, no results using REMIND 3.2 were yet submitted to the ecemf-internal database (Simon Moreno tried yesterday but failed, this PR is fixing the registration so that he can proceed).

File: mappings/REMIND_3.2.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  - CHA: REMIND 3.2|China and Taiwan
+  - IND: REMIND 3.2|India
+  - JPN: REMIND 3.2|Japan
+  - OAS: REMIND 3.2|Other Asia
+  - LAM: REMIND 3.2|Latin America and the Caribbean
+  - MEA: REMIND 3.2|Middle East and North Africa
+  - REF: REMIND 3.2|Russia and Reforming Economies
+  - SSA: REMIND 3.2|Sub-Saharan Africa
+  - USA: REMIND 3.2|United States
+  - DEU: REMIND 3.2|Germany
```
Comment (Renato-Rodrigues) [inline mappings/REMIND_3.2.yml:15]:
```suggestion
    # 12-region version of REMIND
  - EUR: REMIND 3.2|EU 28
  - NEU: REMIND 3.2|Non-EU28 Europe
  # 21-region version of REMIND
  - DEU: REMIND 3.2|Germany
```

File: mappings/REMIND_3.2.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    - CHA
+    - IND
+    - JPN
+    - OAS
+    - LAM
+    - MEA
+    - REF
+    - SSA
+    - USA
+    - DEU
```
Comment (Renato-Rodrigues) [inline mappings/REMIND_3.2.yml:41]:
```suggestion
    # 12-region version of REMIND
    - EUR
    - NEU
    # 21-region version of REMIND
    - DEU
```

File: mappings/REMIND_3.2.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  - Latin America (R5):
+    - LAM
+  - Middle East & Africa (R5):
+    - SSA
+    - MEA
+  - OECD & EU (R5):
+    - JPN
+    - USA
+    - CAZ
+    - DEU
```
Comment (Renato-Rodrigues) [inline mappings/REMIND_3.2.yml:95]:
```suggestion
    # 12-region version of REMIND
    - EUR
    - NEU
    # 21-region version of REMIND
    - DEU
```

File: mappings/REMIND_3.2.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    - ENC
+    - ESC
+    - ESW
+    - EWN
+    - NEN
+    - NES
+
+# aggregations used in the ECEMF project
+  - EU27 & UK:
+    - DEU
```
Comment (Renato-Rodrigues) [inline mappings/REMIND_3.2.yml:55]:
```suggestion
    # 12-region version of REMIND
    - EUR
    # 21-region version of REMIND
    - DEU
```

File: mappings/REMIND_3.2.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    - FRA
+    - UKI
+    - ECE
+    - ECS
+    - ENC
+    - ESC
+    - ESW
+    - EWN
+  - EU27 & UK (*):
+    - DEU
```
Comment (Renato-Rodrigues) [inline mappings/REMIND_3.2.yml:65]:
```suggestion
    # 12-region version of REMIND
    - EUR
    # 21-region version of REMIND
    - DEU
```

**Review by Renato-Rodrigues — CHANGES_REQUESTED** [review]

File: definitions/region/model_native_regions/REMIND_3.2.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  - REMIND 3.2|Russia and Reforming Economies:
+      iso3_codes: [ARM, AZE, BLR, GEO, KAZ, KGZ, MDA, RUS, TKM, UKR, UZB, TJK]
+  - REMIND 3.2|Sub-Saharan Africa:
+      iso3_codes: [AGO, BDI, BEN, BFA, BWA, CAF, CIV, CMR, COD, COG, COM, CPV, DJI,
+        ERI, ETH, GAB, GHA, GIN, GMB, GNB, GNQ, KEN, LBR, LSO, MDG, MLI, MOZ, MRT,
+        MUS, MWI, MYT, NAM, NER, NGA, REU, RWA, SEN, SHN, SLE, SOM, SSD, STP, SWZ,
+        SYC, TGO, TZA, UGA, ZAF, ZMB, ZWE, TCD]
+  - REMIND 3.2|United States:
+      iso3_codes: USA
+  - REMIND 3.2|Germany:
```
Comment (Renato-Rodrigues) [inline definitions/region/model_native_regions/REMIND_3.2.yaml:32]:
```suggestion
# 12-region version of REMIND
  - REMIND 3.2|EU 28:
      iso3_codes: [ALA, AUT, BEL, BGR, CYP, CZE, DNK, ESP, EST, FIN, FRO, GGY, GIB,
        GRC, HRV, HUN, IMN, IRL, ITA, JEY, LTU, LUX, LVA, MLT, NLD, POL, PRT, ROU,
        SVK, SVN, SWE, FRA, DEU, ITA, GIB, GGY, GBR]
  - REMIND 3.2|Non-EU28 Europe:
      iso3_codes: [ALB, AND, BIH, CHE, GRL, ISL, LIE, MCO, MKD, MNE, NOR, SJM, SMR,
        SRB, TUR, VAT]
# 21-region version of REMIND
  - REMIND 3.2|Germany:
```

**Review by Renato-Rodrigues — APPROVED** [review]

**Comment by danielhuppmann** [comment]
Permissions added for the REMIND team to submit scenarios for REMIND 3.2 to the ECEMF-internal Scenario Explorer

**Pull request merged**